src/features/create-dao/Steps/TreasuryForm.tsx

```diff
@@ -1,24 +1,56 @@
 import { FC, useContext } from 'react';
 
+import * as Yup from 'yup';
 import { Form, Formik } from 'formik';
 
 import { Wizard } from '@zero-tech/zui/components';
 import { CreateDAOFormContext } from '../';
 
 import styles from './TreasuryForm.module.scss';
+import { WrappedInput } from '../../ui/WrappedInput/WrappedInput';
+import classNames from 'classnames/bind';
 
-interface TreasuryFormProps {
+const validationSchema = Yup.object().shape({
+	gnosisSafe: Yup.string().required('Gnosis Safe is required'),
+});
+
+export interface TreasuryFormProps {
 	onClose: () => void;
 }
 
+const cx = classNames.bind(styles);
+
 export const TreasuryForm: FC<TreasuryFormProps> = ({ onClose }) => {
 	const { treasury, onTreasurySubmit } = useContext(CreateDAOFormContext);
 
 	return (
-		<Formik initialValues={treasury} onSubmit={onTreasurySubmit}>
-			{({ submitForm }) => (
+		<Formik
+			initialValues={treasury}
+			onSubmit={(values) => onTreasurySubmit(values)}
+			validationSchema={validationSchema}
+		>
+			{({ values, errors, touched, setFieldValue, submitForm }) => (
 				<Form>
-					<p>Enter treasury content here...</p>
+					<div>
+						<WrappedInput
+							label="What Ethereum address do you want all of the DAO’s funds to be held in?"
+							value={values.gnosisSafe}
+							placeholder="Gnosis Safe"
```

Comment:
Not in line with the designs but for consistency with the other forms let's change this to 'Enter gnosis address...'.

Reply:
Updated